sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1530,10 +1530,15 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
   }
 
   static std::once_flag ZeCallCountInitialized;
-  std::call_once(ZeCallCountInitialized, []() {
-    if (ZeDebug & ZE_DEBUG_CALL_COUNT)
-      ZeCallCount = new std::map<const char *, int>;
-  });
+  try {
+    std::call_once(ZeCallCountInitialized, []() {
+      if (ZeDebug & ZE_DEBUG_CALL_COUNT) {
+        ZeCallCount = new std::map<const char *, int>;
+      }
+    });
+  } catch (...) {
+    return PI_OUT_OF_HOST_MEMORY;
+  }
```

Comment:
```suggestion
  } catch (const std::bad_alloc &) {
    return PI_OUT_OF_HOST_MEMORY;
  } catch (...) {
    return PI_ERROR_UNKNOWN;
  }```

Reply:
Your suggested change is on the outdated source code. So, cannot commit.
I pushed this change suggestion.